includes/mailpoet/class-sensei-mailpoet.php

```diff
@@ -0,0 +1,369 @@
+		}
+	}
+
+	/**
+	 * Remove student as subscriber to a course list on MailPoet. A proxy to SenseiMailPoet::remove_student_subscribers.
+	 *
+	 * @param int $user_id User ID.
+	 * @param int $course_id Post ID.
+	 *
+	 * @return void
+	 */
+	public function remove_student_course_subscriber( $user_id, $course_id ) {
+		$this->remove_student_subscribers( $course_id, array( $user_id ) );
+	}
+
+	/**
+	 * Add student as subscriber to a course list on MailPoet. A proxy to SenseiMailPoet::add_student_subscriber.
+	 *
+	 * @param int $user_id User ID.
+	 * @param int $course_id Post ID.
+	 *
+	 * @return void
+	 */
+	public function add_student_course_subscriber( $user_id, $course_id ) {
+		$this->add_student_subscriber( $course_id, $user_id );
+	}
+
+	/**
+	 * Get subscribers of a MailPoet list.
+	 * Returns null if the list does not exist.
+	 *
+	 * @param int $mp_list_id MailPoet list ID.
+	 *
+	 * @return array
+	 */
+	public function get_mailpoet_subscribers( $mp_list_id ) {
+		try {
+			return $this->mailpoet_api->getSubscribers( array( 'listId' => (int) $mp_list_id ) );
+		} catch ( \MailPoet\API\MP\v1\APIException $exception ) {
+			return array();
+		}
+	}
+
+	/**
+	 * Figure out which students to add as subscribers to lists on MailPoet for courses/groups.
+	 *
+	 * @param array      $students A list of students belonging to this group/course list.
+	 * @param string|int $list_id ID of the list on MailPoet.
+	 * @param array      $mp_subscribers A list of subscribers already on MailPoet.
+	 *
+	 * @return void
+	 */
+	public function maybe_add_subscribers( $students, $list_id, $mp_subscribers ) {
+		$options = array(
+			'send_confirmation_email'      => false,
+			'schedule_welcome_email'       => false,
+			'skip_subscriber_notification' => false,
+		);
+		foreach ( $students as $student ) {
+			if ( ! array_key_exists( $student->user_email, $mp_subscribers ) ) {
+				$subscriber_data = array(
+					'email'      => $student->user_email,
+					'first_name' => $student->display_name,
+				);
+
+				try {
+					// All WordPress users are already on a list 'WordPress Users' on MailPoet.
+					$mp_subscriber = $this->mailpoet_api->getSubscriber( $student->user_email );
+					$this->mailpoet_api->subscribeToList( $mp_subscriber['id'], $list_id, $options );
+				} catch ( \MailPoet\API\MP\v1\APIException $exception ) {
+					if ( 4 === $exception->getCode() ) {
+						// subscriber does not exist.
+						$this->mailpoet_api->addSubscriber( $subscriber_data, array( $list_id ), $options );
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Figure out which subscribers to remove from lists on MailPoet for courses/groups.
+	 *
+	 * @param array      $subscribers A list of subscribers already on MailPoet.
+	 * @param string|int $list_id ID of the list on MailPoet.
+	 * @param array      $students A list of students belonging to this group/course list.
+	 *
+	 * @return void
+	 */
+	public function maybe_remove_subscribers( $subscribers, $list_id, $students ) {
```

Comment:
Seems like `maybe_add_subscribers` and `maybe_remove_subscribers` could be better off merged into one method, something like `sync_subscribers` — both for clarity and efficiency.

Reply:
Thanks. Added this and other review feedbacks in https://github.com/Automattic/sensei/pull/6503/commits/457e4ed88238889d32a4fec2780802ce463d2ffb
`sync_subscribers` can be more efficient and I'll still refactor it later.